code/LINDA/LINDA_fire.dm

```diff
@@ -26,13 +26,23 @@
 	if((exposed_temperature > PLASMA_MINIMUM_BURN_TEMPERATURE) && air_contents.toxins > 0.5)
 		igniting = 1
 
+	var/obj/effect/decal/cleanable/liquid_fuel/liquid = locate() in src
```

Comment:
Specifically to accomplish what this is aiming for, I would recommend the following:
Two vars on `/liquid_fuel`
```
/obj/effect/decal/cleanable/liquid_fuel
    var/processing = 0
    var/tick_life = 10 //object ticks before being "consumed"
```
Followed by the fire proc:
```
/obj/effect/decal/cleanable/liquid_fuel/fire_act()
    if(processing)
        return
    processing = 1
    processing_objects |= src
```
Followed by it's process
```
/obj/effect/decal/cleanable/liquid_fuel/process()
    tick_life--
    atmos_spawn_air(SPAWN_HEAT|SPAWN_TOXINS, 200)
    for(var/obj/effect/decal/cleanable/liquid_fuel/L in loc)
        tick_life += L.tick_life
        qdel(L)
    if(tick_life < 0)
        qdel(src)
```